contracts/staking/StakingV1_1.sol

```diff
@@ -63,13 +143,31 @@ contract StakingV1_1 is StakingState, StakingConstants {
 
             IERC20(token).safeTransferFrom(msg.sender, address(this), stakeAmount);
 
+            //Deposit to sushi masterchef
+            if(token == LPToken){
+                uint256 sushiBalanceBefore = IERC20(SUSHI).balanceOf(address(this));
+                IMasterChefSushi(SUSHI_MASTERCHEF).deposit(
+                    BZRX_ETH_SUSHI_MASTERCHEF_PID,
+                    IERC20(LPToken).balanceOf(address(this))
+                );
+                uint256 sushiRewards = IERC20(SUSHI).balanceOf(address(this)) - sushiBalanceBefore;
+                if(sushiRewards != 0){
+                    _addAltRewards(SUSHI, sushiRewards);
+                }
+            }
             emit Stake(
                 msg.sender,
                 token,
                 msg.sender, //currentDelegate,
                 stakeAmount
             );
         }
+
+        _paySushiRewards(
```

Comment:
since paySushi rewards should happen only for LP stakers. should this be moved inside a loop where `token == LPToken` . then it will be called only to LP Stakers and not everybody else

Reply:
and maybe have a separate function to claim them

Reply:
nope, we send sushi if user has pending sushi even if he didnt stake lp at the moment but had lp balance